src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -43,42 +57,112 @@ public Intake(int motorId, int frontPiston, int backPiston) {
-        intakeController = intakeMotor.getClosedLoopController();
-        intakeEncoder.setPosition(0);
+        grabMotor.configure(intakeConfig, ResetMode.kNoResetSafeParameters, PersistMode.kPersistParameters);
+        grabEncoder = grabMotor.getEncoder();
+        grabController = grabMotor.getClosedLoopController();
+        grabEncoder.setPosition(0);
+
+        intakeConfig.encoder.positionConversionFactor(Constants.IntakeConstants.TILT_MOTOR_COEFFICIENT); // temporary TODO
+        tiltMotor.configure(intakeConfig, ResetMode.kNoResetSafeParameters, PersistMode.kPersistParameters);
+        tiltEncoder = tiltMotor.getEncoder();
+        tiltController = tiltMotor.getClosedLoopController();
+        tiltEncoder.setPosition(0);
+
+
+        SmartDashboard.putBoolean("Passive intake out", false);
+    }
+
+    public void prioState() {
+        if (SmartDashboard.getBoolean("Passive intake out", false) & passiveIntakePrio == PassiveIntake.IN) {
+            passiveIntakePrio = PassiveIntake.OUT;
+        }
+        else if (SmartDashboard.getBoolean("Passive intake out", true) & passiveIntakePrio == PassiveIntake.OUT){
+            passiveIntakePrio = PassiveIntake.IN;
+        }
+    }
+
+    public SequentialCommandGroup grabCoralSequence() {
+        return new SequentialCommandGroup(
+            intakeOut(), grabCoral()
+        );
+    }
 
-        intakeModule = new PneumaticsControlModule(0);
+    public SequentialCommandGroup releaseCoralSequence() {
+        return new SequentialCommandGroup(
+           stopGrab(), intakeIn(), releaseCoral(), new WaitCommand(.5), stopRelease()
+        );
+    }
+    public  Command intakeMid() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_MID, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.MID;
+        });
+    }
 
-        intakeSolenoid = new Solenoid(0, PneumaticsModuleType.CTREPCM, frontPiston);
-        intakeSolenoid.set(false); //might be needed to be set to true
+    public Command intakeOut() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_OUT, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.OUT;
+        });
     }
 
-    public Command toggleSolenoid() {
-        return Commands.runOnce(() -> {
-            if (solenoidState == SolenoidState.FALSE) {
-                intakeSolenoid.set(true);
-                solenoidState = SolenoidState.TRUE;
-            } else {
-                intakeSolenoid.set(false);
-                solenoidState = SolenoidState.FALSE;
+    public Command intakeIn() {
+        return Commands.runOnce(()-> {
+            tiltController.setReference(Constants.IntakeConstants.EXTENSION_IN, SparkMax.ControlType.kPosition);
+            intakePosition = IntakePosition.IN;
+        });
+    }
+    
+    public Command stopGrab() {
```

Comment:
Takto souhlasím, ale místo toho, aby se to vyplo hned, jak driver pustí tlačítko, tak by to chtělo projet následující proces:
Je zmáčknuté první tlačítko? -> nevypínat
Je zmáčknuté druhé tlačítko? -> vypnout
Není zmáčknuté žádné tlačítko? -> vypnout

Tohle jde řešit přes Trigger strukturu (jakmile dojde k nějaké změně v Trigger objektu, spustí se Command)